server/src/Persistence.hs

```diff
@@ -69,9 +76,94 @@ apply op val = do
 
 -- * IO
 
+loadFromBackend :: StorageBackend -> PersistenceConfig -> IO (Either String PersistentValue)
+loadFromBackend File pc = loadFile pc
+loadFromBackend Sqlite pc = loadSqliteFile pc
+
+syncToBackend :: StorageBackend -> PersistentValue -> IO ()
+syncToBackend File pv = syncFile pv
+syncToBackend Sqlite pv = syncSqliteFile pv
+
+-- * SQLite loading and syncing
+
+-- | There is currently just one row and it contains only one column of type SBS.ByteString.
+-- This single field holds the whole JSON value for now.
+data JsonRow = JsonRow {jsonByteString :: SBS.ByteString} deriving (Show)
+
+instance FromRow JsonRow where
+  fromRow = JsonRow <$> field
+
+loadSqliteFile :: PersistenceConfig -> IO (Either String PersistentValue)
+loadSqliteFile config = runExceptT $ do
+  liftIO $ forM_ (pcMetrics config) $ \m -> do
+    size <- getFileSize (pcDataFile config)
+    Metrics.setDataSize size m
+
+  value <- readSqliteData (pcDataFile config) (pcLogger config)
+  valueVar <- lift $ newTVarIO value
+  dirtyVar <- lift $ newTVarIO False
+  journal <- for (pcJournalFile config) openJournal
+  let val = PersistentValue
+        { pvConfig  = config
+        , pvValue   = valueVar
+        , pvIsDirty = dirtyVar
+        , pvJournal = journal
+        }
+  recoverJournal val
+  return val
+
+-- | Read and decode the Sqlite data file from disk
+readSqliteData :: FilePath -> Logger -> ExceptT String IO Store.Value
+readSqliteData filePath _logger = ExceptT $ do
+  -- read the data from SQLite
+  conn <- liftIO $ open filePath
+  -- TODO: We might want to have a primary key here after all. Then we could have multiple
+  -- JSON values next to each other.
+  liftIO $ execute_ conn "CREATE TABLE IF NOT EXISTS test (value BLOB)"
```

Comment:
It's a bit unexpected that read creates the table. I would expect reads to work against a read-only database, for example. Maybe we should move this into `loadSqliteFile` instead?

Reply:
I have added a `setupStorageBackend` function now, that is called at program start and that ensures that we can access the file, and that validates that it has the right format. If there is an error, it prints a nice error message and exits the program.